incubator/hnc/pkg/controllers/config_controller.go

```diff
@@ -0,0 +1,109 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	api "github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/api/v1alpha1"
+	"github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/pkg/config"
+)
+
+// ConfigReconciler is responsible for determining the HNC configuration from the HNCConfiguration CR,
+// as well as ensuring all objects are propagated correctly when the HNC configuration changes.
+// It can also set the status of the HNCConfiguration CR.
+type ConfigReconciler struct {
+	client.Client
+	Log logr.Logger
+}
+
+// Reconcile sets up some basic variable and logs the Spec.
+// TODO: Updates the comment above when adding more logic to the Reconcile method.
+func (r *ConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	// TODO: Surface the error more prominently and add a validating admission controller to prevent
+	// the problem in the first place.
+	if req.NamespacedName.Name != api.HNCConfigSingleton {
+		r.Log.Error(nil, "Singleton name is wrong. It should be 'config'.")
+		return ctrl.Result{}, nil
+	}
+
+	ctx := context.Background()
+	inst, err := r.getSingleton(ctx)
+	if err != nil {
+		r.Log.Error(err, "Couldn't read singleton.")
+		return ctrl.Result{}, nil
+	}
+
+	r.Log.Info("Reconciling cluster-wide HNC configuration.")
```

Comment:
Add "TODO" here to actually modify the other reconcilers

Reply:
Done. Feel free to take a look.

Reply:
lgtm